### PR TITLE
AML cloud support

### PR DESCRIPTION
**Pull request by friggog**
Adds support for running simulations on AML, addressing #40 

TODO:
- [x] Needs #69 for this to work when running from a clone of the repo
- [x] Hyperdrive not currently supported
- [x] Figure out outputs - what is the typical output we want/how do we transfer back from AML?
- ~~_No sysimage_ - docker has packages built/precompiled when using OpticSim as a dependency which is probably sufficient~~
- [x] Add docs

We could call julia directly using ScriptRunConfig "command" rather than having the entry script stuff, I think we'll always need to resolve the environment first though and it becomes a pain to configure thread count.

### CODE REVIEW

**Comment by alfredclwong** [comment]
re outputs: I was thinking of writing some kind of "Ray Database" type, similar to a .ZRD file in Zemax, maybe next week. Any other thoughts?

**Review by BrianGun — COMMENTED** [review]
@friggog do you have a simple example you could add to the Examples directory? Maybe a call to traceMT which returns an image? Since any example would need an Azure account to bill against that part could be left for the user to fill in, perhaps with inline comments that explain what the user needs to do to make it work. And maybe a paragraph of documentation in the docs? You could add a new section, Cloud Computation, or something like that. Otherwise people are going to have a hard time figuring out how to use this.

**Comment by BrianGun** [comment]
@friggog were you going to add an example and some documentation? It will be confusing to new users otherwise.

**Comment by friggog** [comment]
Yeah there are a few things to do still (see list on top comment here) hopefully will get around to it this week but will see how busy I am

**Comment by BrianGun** [comment]
Charlie,

You should have permission to approve pull requests. Could you verify for me?

-brian

**Comment by friggog** [comment]
@alfredclwong can I preview built docs from this branch somehow?

**Comment by alfredclwong** [comment]
@friggog deploydocs has an argument that'll push PR previews to the gh-pages branch. i've pushed a commit that'll try out that feature.

**Review by alfredclwong — APPROVED** [review]
Thanks for taking the time to do this. I don't think I have the permissions yet to try it out, but I assume you've tested it yourself.

File: docs/src/cloud.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+- Subscription ID, of the form `XXXXXXXX-XXX-XXX-XXXX-XXXXXXXXXXXX`
+- Resource group name
+- Workspace name
+- Compute cluster name
+
+This information can be cached either to a specific file, or globally:
+
+```@docs
+Optics.Cloud.cache_run_config
```
Comment (alfredclwong) [inline docs/src/cloud.md:20]: Need to change Optics -> OpticSim

**Review by alfredclwong — CHANGES_REQUESTED** [review]
3 lines need changing - I'll push the changes in a sec

File: docs/src/cloud.md (hunk 1 above)
Comment (friggog) [reply docs/src/cloud.md:20]: Yeah, publishing works great - definitely a useful feature https://microsoft.github.io/OpticSim.jl/previews/PR71/cloud/

**Comment by friggog** [comment]
> Thanks for taking the time to do this. I don't think I have the permissions yet to try it out, but I assume you've tested it yourself.

yeah it seems to be working for a simple test script. The pycall setup is the most dodgy bit so maybe need some refinement there, seems to be very dependent on local environment (ie conda/python install)

I might be able to give you permissions though I just removed my ownership so we might need Ran or Brian to do that now

**Review by alfredclwong — APPROVED** [review]

**Pull request merged**